tests/unit/wallet/test_encryption.py

```diff
@@ -0,0 +1,108 @@
+            'name': 'deterministic-chain',
+            'receiving': {'gap': 17, 'maximum_uses_per_address': 2},
+            'change': {'gap': 10, 'maximum_uses_per_address': 2}
+        }
+    }
+
+    def setUp(self):
+        self.ledger = MainNetLedger({
+            'db': WalletDatabase(':memory:'),
+            'headers': Headers(':memory:')
+        })
+        return self.ledger.db.open()
+
+    def test_encrypt_wallet(self):
+        account = Account.from_dict(self.ledger, Wallet(), self.unencrypted_account)
+        account.encryption_init_vector = self.init_vector
+
+        self.assertFalse(account.serialize_encrypted)
+        self.assertFalse(account.encrypted)
+        account.encrypt(self.password)
+        self.assertFalse(account.serialize_encrypted)
+        self.assertTrue(account.encrypted)
+
+        self.assertEqual(account.seed, self.encrypted_account['seed'])
+        self.assertEqual(account.private_key, self.encrypted_account['private_key'])
+
+        self.assertEqual(account.to_dict()['seed'], self.encrypted_account['seed'])
+        self.assertEqual(account.to_dict()['private_key'], self.encrypted_account['private_key'])
+
+        account.serialize_encrypted = True
+        account.decrypt(self.password)
+
+        self.assertEqual(account.seed, self.unencrypted_account['seed'])
+        self.assertEqual(account.private_key.extended_key_string(), self.unencrypted_account['private_key'])
+
+        self.assertEqual(account.to_dict()['seed'], self.encrypted_account['seed'])
+        self.assertEqual(account.to_dict()['private_key'], self.encrypted_account['private_key'])
+
+        account.encryption_init_vector = None
+        self.assertNotEqual(account.to_dict()['seed'], self.encrypted_account['seed'])
+        self.assertNotEqual(account.to_dict()['private_key'], self.encrypted_account['private_key'])
+
+        self.assertFalse(account.encrypted)
+        self.assertTrue(account.serialize_encrypted)
+
+    def test_decrypt_wallet(self):
+        account = Account.from_dict(self.ledger, Wallet(), self.encrypted_account)
+        account.encryption_init_vector = self.init_vector
+
+        self.assertTrue(account.encrypted)
+        self.assertTrue(account.serialize_encrypted)
+        account.decrypt(self.password)
+        self.assertFalse(account.encrypted)
+        self.assertTrue(account.serialize_encrypted)
+
+        self.assertEqual(account.seed, self.unencrypted_account['seed'])
+        self.assertEqual(account.private_key.extended_key_string(), self.unencrypted_account['private_key'])
+
+        self.assertEqual(account.to_dict()['seed'], self.encrypted_account['seed'])
+        self.assertEqual(account.to_dict()['private_key'], self.encrypted_account['private_key'])
+
+        account.serialize_encrypted = False
+        self.assertEqual(account.to_dict()['seed'], self.unencrypted_account['seed'])
+        self.assertEqual(account.to_dict()['private_key'], self.unencrypted_account['private_key'])
```

Comment:
Is this duplicate of what you added to torba? I would drop it from lbrynet unless there are lbrynet specific things it needs to ensure. I'd rather rely on integration tests that use the Daemon encrypt/decrypt API for sanity testing and leave the unit tests in torba.